examples/LLM/llama2/chat_app/torchserve_server_app.py

```diff
@@ -0,0 +1,200 @@
+import json
+import os
+
+import requests
+import streamlit as st
+
+MODEL_NAME = "llamacpp"
+# App title
+st.set_page_config(page_title="🦙💬 Llama 2 TorchServe Serve")
+
+
+def start_server():
+    os.system("torchserve --start --model-store model_store --ncs")
+    st.session_state.started = True
+    st.session_state.stopped = False
+    st.session_state.registered = False
+
+
+def stop_server():
+    os.system("torchserve --stop")
+    st.session_state.stopped = True
+    st.session_state.started = False
+    st.session_state.registered = False
+
+
+def _register_model(url):
+    res = requests.post(url)
+    if res.status_code != 200:
+        server_state_container.error("Error registering model", icon="🚫")
+        st.session_state.started = True
+        return
+    st.session_state.registered = True
+    st.session_state.started = False
+    st.session_state.stopped = False
+    server_state_container.caption(res.text)
+
+
+def register_model():
+    if not st.session_state.started:
+        server_state_container.caption("TorchServe has stopped. Start it")
+        return
+    url = (
+        "http://localhost:8081/models?model_name="
+        + MODEL_NAME
+        + "&url="
+        + MODEL_NAME
+        + ".tar.gz"
+        + "&initial_workers=1&synchronous=true"
+    )
+    _register_model(url)
+
+
+def get_status():
+    if st.session_state.registered:
+        url = "http://localhost:8081/models/" + MODEL_NAME
+        res = requests.get(url)
+        if res.status_code != 200:
+            model_state_container.error("Error getting model status", icon="🚫")
```

Comment:
so hopefully actual error logs are still somewhere? Don't wanna swallow the actual error message for someone trying to debug this

Reply:
Yes, the terminal where you start the server shows the actual logs. But let me add a comment in the readme